apis/v1alpha1/greptimedbcluster_types.go

```diff
@@ -926,6 +944,22 @@ type GreptimeDBCluster struct {
 	Status GreptimeDBClusterStatus `json:"status,omitempty"`
 }
 
+func (in *GreptimeDBCluster) GetDatanodeReplicas() int32 {
+	var count int32
+
+	if replicas := in.GetDatanode().GetReplicas(); replicas != nil {
+		return *replicas
+	}
```

Comment:
If datanode and datanodeGroups exist at the same time, only return datanode replicas here, not have datanodeGroups replicas.

Maybe have two function (GetDatanodeReplicas and GetDatanodeGroupsReplicas):
```
func (in *GreptimeDBCluster) GetDatanodeReplicas() int32 {
	if replicas := in.GetDatanode().GetReplicas(); replicas != nil {
		return *replicas
	}
}	
```
```
func (in *GreptimeDBCluster) GetDatanodeGroupsReplicas() int32 {
  var count int32
  for _, datanodeGroup := range in.GetDatanodeGroups() {
		if replicas := datanodeGroup.GetReplicas(); replicas != nil {
			count += *replicas
		}
	}

	return count  
}	
```

Reply:
It's unexpected behavior. `datanode` and `datanodeGroups` should not be set at the same time. I add the validation rule to stop it:

```
	if in.GetDatanode() != nil && len(in.GetDatanodeGroups()) > 0 {
		return fmt.Errorf("datanode and datanodeGroups cannot be set at the same time")
	}
```